pulsar-transformations/src/test/java/com/datastax/oss/pulsar/functions/transforms/TransformFunctionTest.java

```diff
@@ -127,6 +142,90 @@ void testDropFields() throws Exception {
     assertNull(valueAvroRecord.getSchema().getField("valueField3"));
   }
 
+  @Test
+  void testMatchingPredicate() throws Exception {
+    String userConfig =
+        (""
+            + "{\"steps\": ["
+            + "    {\"type\": \"drop-fields\", \"fields\": \"keyField1\", \"filter\": \"key.keyField1='key1'\"},"
+            + "    {\"type\": \"drop-fields\", \"fields\": \"keyField2\", \"filter\": \"key.keyField2='key2'\"}"
+            + "]}");
+    Map<String, Object> config =
+        new Gson().fromJson(userConfig, new TypeToken<Map<String, Object>>() {}.getType());
+    TransformFunction transformFunction = new TransformFunction();
+
+    Record<GenericObject> record = Utils.createTestAvroKeyValueRecord();
+    Utils.TestContext context = new Utils.TestContext(record, config);
+    transformFunction.initialize(context);
+    transformFunction.process(record.getValue(), context);
+
+    Utils.TestTypedMessageBuilder<?> message = context.getOutputMessage();
```

Comment:
This doesn't work anymore. Need to work on the Record returned by `transformFunction.process`